Plots/MapProjections/NCL_sat_1.py

```diff
@@ -0,0 +1,380 @@
+        transform (:class:`cartopy._crs`):
+            Instance of CRS that represents the source coordinate system of coordinates.
+            (ex. ccrs.Geodetic()).
+        ax (:class:`matplotlib.pyplot.axis`):
+            Axis containing the contour set.
+        proj (:class:`cartopy.crs`):
+            Projection 'ax' is defined by.
+            This is the instaance of CRS that the coordinates will be transformed to.
+        Clevels (:class:`list`):
+            List of coordinate tuples in GPS form (lon in degrees, lat in degrees)
+            that specify where the contours with regular field variable values should be plotted.
+        lowClevels (:class:`list`):
+            List of coordinate tuples in GPS form (lon in degrees, lat in degrees)
+            that specify where the contours with low field variable values should be plotted.
+        highClevels (:class:`list`):
+            List of coordinate tuples in GPS form (lon in degrees, lat in degrees)
+            that specify where the contours with high field variable values should be plotted.
+        rfs (:class:`int`):
+            Font size of regular contour labels.
+        efs (:class:`int`):
+            Font size of extrema contour labels.
+        rHorizontal (:class:`bool`):
+            Setting this to "True" will cause the regular contour labels to be horizontal.
+        eHorizontal (:class:`bool`):
+            Setting this to "True" will cause the extrema contour labels to be horizontal.
+
+        whitebbox (:class:`bool`):
+            Setting this to "True" will cause all labels to be plotted with white backgrounds
+    Returns:
+        allLabels (:class:`list`):
+            List of text instances of all contour labels
+    """
+
+    import numpy as np
+    import matplotlib.pyplot as plt
+
+    # Create array of coordinates in the same shape as field variable data
+    # so each coordinate can be easily mapped to its data value.
+    coordarr = []
+    for y in np.array(da.lat):
+        temparr = []
+        for x in np.array(da.lon):
+            temparr.append((x, y))
+        coordarr.append(temparr)
+    coordarr = np.array(coordarr)
+
+    # Initialize empty array that will be filled with contour label text objects and returned
+    allLabels = []
+
+    # Plot any regular contour levels
+    if Clevels != []:
+        clevelpoints = proj.transform_points(transform,
+                                             np.array([x[0] for x in Clevels]),
+                                             np.array([x[1] for x in Clevels]))
+        transformedClevels = [(x[0], x[1]) for x in clevelpoints]
+        ax.clabel(contours, manual=transformedClevels, inline=True, fontsize=rfs, colors='k', fmt="%.0f")
+        [allLabels.append(txt) for txt in contours.labelTexts]
+        if rHorizontal is True:
+            [txt.set_rotation('horizontal') for txt in contours.labelTexts]
+
+    # Plot any low contour levels
+    if lowClevels != []:
```

Comment:
Could this block of code and the next be made into a function that plots either highs or lows? It seems like code is duplicated, the only real difference being the label that is used.